src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -29,43 +30,55 @@ public static IServiceCollection AddBlobDataStore(this IServiceCollection servic
+        }
 
-            services.Add<BlobClientProvider>()
-                .Singleton()
-                .AsSelf()
-                .AsService<IHostedService>() // so that it starts initializing ASAP
-                .AsService<IRequireInitializationOnFirstRequest>(); // so that web requests block on its initialization.
+        /// <summary>
+        /// Adds a singleton <see cref="BlobServiceClient"/> to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <returns>A <see cref="BlobClientBuilder"/> for configuring the <see cref="BlobServiceClient"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="services"/> is <see langword="null"/>.</exception>
+        public static BlobClientBuilder AddBlobServiceClient(this IServiceCollection services)
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
 
-            services.Add(sp => sp.GetService<BlobClientProvider>().CreateBlobClient())
-                .Singleton()
-                .AsSelf();
+            services.AddOptions();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<BlobDataStoreConfiguration>>(new DefaultBlobDataStoreConfiguration()));
+            services.TryAddSingleton(p => BlobClientFactory.Create(p.GetRequiredService<IOptions<BlobDataStoreConfiguration>>().Value));
 
-            services.Add<BlobClientReadWriteTestProvider>()
-                .Singleton()
-                .AsService<IBlobClientTestProvider>();
+            return new BlobClientBuilder(services);
+        }
 
-            services.Add<BlobClientInitializer>()
-                .Singleton()
-                .AsService<IBlobClientInitializer>();
+        /// <summary>
+        /// Adds a singleton <see cref="BlobServiceClient"/> to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <param name="configure">A delegate for configuring the <see cref="BlobDataStoreConfiguration"/>.</param>
+        /// <returns>A <see cref="BlobClientBuilder"/> for configuring the <see cref="BlobServiceClient"/>.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="services"/> or <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public static BlobClientBuilder AddBlobServiceClient(this IServiceCollection services, Action<BlobDataStoreConfiguration> configure)
+        {
+            EnsureArg.IsNotNull(configure, nameof(configure));
 
-            services.TryAddSingleton<RecyclableMemoryStreamManager>();
+            BlobClientBuilder builder = services.AddBlobServiceClient();
+            services.Configure(configure);
 
-            return services;
+            return builder;
         }
```

Comment:
There are so many public methods. It will be confusing to know which one to use. 

All we need is 2 right. Both need to initialize everything except in different workflow